### PR TITLE
Fix: MySQL integration tests

### PR DESCRIPTION
**Pull request by Klaas-Ritense**

### CODE REVIEW

File: contract/src/main/java/com/ritense/valtimo/contract/database/MysqlQueryDialectHelper.java [hunk 1]
```diff
@@ -48,29 +48,43 @@ public <T> Expression<T> getJsonValueExpression(CriteriaBuilder cb, Path column,
 
     @Override
     public Predicate getJsonValueExistsExpression(CriteriaBuilder cb, Path column, String value) {
+        Expression<?> searchColumn = column;
+        Expression<?> searchValue = cb.literal("%" + value.trim() + "%");
+        if (column.getJavaType() == String.class) {
+            searchColumn = cb.function(LOWER_CASE_FUNCTION, String.class, searchColumn);
+            searchValue = cb.function(LOWER_CASE_FUNCTION, String.class, searchValue);
```
Comment (Klaas-Ritense) [inline contract/src/main/java/com/ritense/valtimo/contract/database/MysqlQueryDialectHelper.java:55]: Mysql `lower()` fails on a column with type 'json'.

Don't know why it didn't fail before

File: document/src/main/resources/config/liquibase/changelog/20240201-add-internal-case-status-table.xml [hunk 2]
```diff
@@ -49,6 +49,13 @@
...
 
+    <changeSet id="2.5" author="Ritense">
+        <createIndex indexName="idx_internal_case_status_case_name_status_key" tableName="internal_case_status" unique="true">
+            <column name="case_definition_name"/>
+            <column name="internal_case_status_key"/>
+        </createIndex>
+    </changeSet>
+
     <changeSet author="Ritense" id="3">
         <addForeignKeyConstraint
```
Comment (Klaas-Ritense) [inline document/src/main/resources/config/liquibase/changelog/20240201-add-internal-case-status-table.xml:60]: This changeset fails without an index on `referencedColumnNames="case_definition_name, internal_case_status_key"`

File: form/src/main/java/com/ritense/form/repository/FormDefinitionRepository.java [hunk 3]
```diff
@@ -35,5 +37,6 @@ public interface FormDefinitionRepository extends JpaRepository<FormIoFormDefini
 
     Optional<FormIoFormDefinition> findByNameIgnoreCase(String name);
 
-    Page<FormDefinition> findAllByNameContainingIgnoreCase(String name, Pageable pageable);
+    @Query("SELECT f FROM FormIoFormDefinition f WHERE upper(f.name) LIKE upper(concat('%', :name, '%'))")
+    Page<FormDefinition> findAllByNameContainingIgnoreCase(@Param("name") String name, Pageable pageable);
```
Comment (Klaas-Ritense) [inline form/src/main/java/com/ritense/form/repository/FormDefinitionRepository.java:41]: Fix for a possible bug in JPA

File: zgw/zaken-api/src/main/resources/config/liquibase/changelog/20240131-zaaktype-link-domain-changelog.xml [hunk 4]
```diff
@@ -35,7 +36,7 @@
             <column name="zaak_type_url" type="VARCHAR(512)">
                 <constraints nullable="false"/>
             </column>
-            <column name="service_task_handlers" type="${jsonType}" defaultValue="[]">
+            <column name="service_task_handlers" type="${jsonType}">
```
Comment (Klaas-Ritense) [inline zgw/zaken-api/src/main/resources/config/liquibase/changelog/20240131-zaaktype-link-domain-changelog.xml:39]: defaultValue isn't allowed for a column with type 'json' on mysql

**Review by theo-ritense — APPROVED** [review]

**Pull request merged**